src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -1047,6 +1067,99 @@ private void EndStartupOperation(ScriptHostStartupOperation operation)
+                ForceFlush(loggerFactory);
+                // Give some time for the logger to flush
+                await Task.Delay(4000);
+            }
+        }
+
+        private ILoggerFactory BuildOtelLoggerFactory()
+        {
+            var appInsightsConnStr = GetConfigurationValue(AppInsightsConnectionString, _config);
+            var otlpEndpoint = GetConfigurationValue(OtlpEndpoint, _config);
+
+            if (appInsightsConnStr is not { Length: > 0 } && otlpEndpoint is not { Length: > 0 })
+            {
+                return null; // Nothing configured
+            }
+
+            // Create a minimal logger factory with OpenTelemetry and Azure Monitor exporter
+            return LoggerFactory.Create(builder =>
+            {
+                builder.AddOpenTelemetry(logging =>
+                {
+                    logging.IncludeScopes = true;
+                    logging.IncludeFormattedMessage = true;
+
+                    if (appInsightsConnStr is { Length: > 0 })
+                    {
+                        logging.AddAzureMonitorLogExporter(options =>
+                        {
+                            options.ConnectionString = appInsightsConnStr;
+
+                            var appInsightsAuthStr = GetConfigurationValue(AppInsightsAuthenticationString, _config);
+                            if (appInsightsAuthStr is { Length: > 0 })
+                            {
+                                var credOptions = AppInsightsCredentialOptions.ParseAuthenticationString(appInsightsAuthStr);
+                                options.Credential = new ManagedIdentityCredential(credOptions.ClientId);
+                            }
+                        });
+                    }
+
+                    if (otlpEndpoint is { Length: > 0 })
+                    {
+                        logging.AddOtlpExporter();
+                    }
+                });
+            });
+        }
+
+        private void ForceFlush(ILoggerFactory loggerFactory)
+        {
+            var serviceProvider = (IServiceProvider)loggerFactory.GetType()
+            .GetField("_serviceProvider", BindingFlags.NonPublic | BindingFlags.Instance)
+            .GetValue(loggerFactory);
+
+            // Get all logger providers from the service provider
+            var providers = serviceProvider?.GetServices<ILoggerProvider>() ?? Enumerable.Empty<ILoggerProvider>();
+
+            foreach (var provider in providers)
+            {
+                if (provider is OpenTelemetryLoggerProvider otelProvider)
+                {
+                    otelProvider.Dispose();
+                }
+            }
+        }
+
+        private static string GetConfigurationValue(string key, IConfiguration configuration = null)
+        {
+            if (configuration != null && configuration[key] is string configValue)
```

Comment:
nit: Can make it even more concise `return configuration?[key] ?? Environment.GetEnvironmentVariable(key);`